### PR TITLE
rename `preferred_optimize_mode` to better reflect its usage

### PR DESCRIPTION
**Pull request by xdBronch**
ive seen this trip up people a good few times, see #20245. i dont think the current name makes much sense.

### CODE REVIEW

**Review by nektro — APPROVED** [review]

File: lib/std/Build.zig [hunk 1]
```diff
@@ -1286,11 +1286,14 @@ pub fn step(b: *Build, name: []const u8, description: []const u8) *Step {
...
 
 pub const StandardOptimizeOptionOptions = struct {
-    preferred_optimize_mode: ?std.builtin.OptimizeMode = null,
+    /// When set, this option causes the standard optimize option to change to a simple boolean flag, `-Drelease`,
+    /// choosing between Debug mode and the mode given here.
+    /// Note that this does not change the default optimization mode.
+    preferred_release_mode: ?std.builtin.OptimizeMode = null,
 };
 
 pub fn standardOptimizeOption(b: *Build, options: StandardOptimizeOptionOptions) std.builtin.OptimizeMode {
```
Comment (alexrp) [inline lib/std/Build.zig:1295]: Maybe panic early here if `preferred_release_mode == .Debug`?

Comment (xdBronch) [reply lib/std/Build.zig:1295]: hmmm, it definitely wouldn't make much sense for someone to do that but I'm not sure I see too much reason to disallow it? I have no strong opinion either way

Comment (nektro) [reply lib/std/Build.zig:1295]: yeah that seems arbitrary for something unlikely to actualize

Comment (alexrp) [reply lib/std/Build.zig:1295]: We're specifically renaming this field from `preferred_optimize_mode` (a name that associates it with the `-O` flag, which allows release modes *and* `Debug`) to `preferred_release_mode`, a name that supposedly better suits its intended purpose. But since it's typed as `?std.builtin.OptimizeMode`, it still allows `Debug`. That means we're trading one kind of confusion for another.

Additionally, given that this is a 'standard option' that's supposed to represent Zig good practice, I think we can very reasonably take the opinionated stance that nobody should make their build script support `-Drelease` and have it mean a debug build. Or if they want to, they have to go out of their way.

In other words: Invalid/nonsensical states shouldn't be representable. Using an inline `enum { ReleaseFast, ReleaseSmall, ReleaseSafe }` and translating to `OptimizeMode` on return would be the 'right' way to do that, but I acknowledge that it might be useful to be able to pass in an existing `OptimizeMode` value that's on hand, hence the panic suggestion.

**Comment by xdBronch** [comment]
i can go ahead and change it to assert now if you dont have any further opinion @alexrp

**Comment by alexrp** [comment]
I think it'd be good to wait for Andrew's thoughts on my suggestion above since it's a question of design philosophy.

**Comment by andrewrk** [comment]
Related:  #19732

**Comment by xdBronch** [comment]
any chance this could get a look before 0.14? ive seen another ~6 cases of this flag confusing people just since the last comment here.

**Comment by andrewrk** [comment]
Thanks for sending this patch.

I'm not sure how this API is going to end up. I don't think status quo will be the final iteration, but I also don't think that your patch will be the final iteration, and I'd like to avoid breaking it multiple times. So, I'm going to close this while we continue to ruminate.

Would you mind opening an issue to track this problem?
